### PR TITLE
Performance improvements

### PR DESCRIPTION
**Pull request by irh**
Hi 👋, I'm working on a project that needs RF64 support and I found this crate, thanks for sharing it!

This PR contains a number of project cleanups and performance improvements; in tests I managed to get roughly ~850x speed improvements in the `wave_deinter.rs` example, with the biggest win coming from reading/writing samples in batches. Hopefully this addresses #13.

Some notes about the PR:
* I was following what clippy suggested in the clippy commits but if any of the changes aren't to your liking then I've kept them in separate commits so they'll be easy to revert if needed.
* The read/write integer/float frame methods have been replaced with generic methods that will convert from/to the provided buffer's sample type. 
  * This is a breaking change, I think it makes sense to go this way but I'm open to alternatives.
* A dependency on [`dasp_sample`](https://docs.rs/dasp/latest/dasp/sample/index.html) has been introduced to simplify the conversion logic.

Please let me know what you think. I know the PR's quite large, I'd be happy to break it up into chunks if it's too much.

### CODE REVIEW

**Comment by iluvcapra** [comment]
Thank you very much for this! This is something I've been meaning to look at for some time just never seem to get around. I'll take a look at the changes this weekend and I may have some comments.

File: src/wavereader.rs [hunk 1]
```diff
@@ -88,78 +90,79 @@ impl<R: Read + Seek> AudioFrameReader<R> {
...
+    F: FnMut() -> Result<S, Error>,
+{
+    for output in buffer.iter_mut().take(sample_count) {
+        *output = read_fn()?;
     }
+
+    Ok(())
 }
 
 /// Wave, Broadcast-WAV and RF64/BW64 parser/reader.
```
Comment (iluvcapra) [inline src/wavereader.rs:168]: Please reactivate this test.

File: tests/integration_test.rs
Comment (iluvcapra) [inline tests/integration_test.rs:1]: Please reactivate all of these tests.

**Review by iluvcapra — DISMISSED** [review]
This all looks good, I'm going to look at clippy configuration options some more, and I'd like all of the deactivated tests re-activated.

The way the test media is supposed to work is you clone the test/test_media.tgz file off git and then inflate it with create_test_media.sh before running your tests, if we can automate this into the cargo test process that would be ideal but I'm not sure exactly how to do that.

File: Cargo.toml
Comment (iluvcapra) [inline Cargo.toml:1]: Also please add yourself as an author.

**Comment by irh** [comment]
Thanks for reviewing the PR so quickly, and sorry for my slow response.

> ...and then inflate it with create_test_media.sh before running your tests

Good grief I don't know how I missed that, sorry for the noise! 

> if we can automate this into the cargo test process that would be ideal but I'm not sure exactly how to do that.

I guess a `build.rs` script could be used for this, although I think expecting people to read the README should be sufficient 😆

**Pull request merged**